### PR TITLE
Expose defaultRuntimeClass

### PR DESCRIPTION
**Pull request by tiraboschi**
**What this PR does / why we need it**:
Expose a knob to let the cluster admin
set a default RuntimeClass for all the VMIs pods.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2203291

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
https://issues.redhat.com/browse/CNV-28792
```

**Release note**:

```release-note
expose defaultRuntimeClass to let the cluster admin set a default RuntimeClass for VMIs pods
```

### CODE REVIEW

**Comment by tiraboschi** [comment]
/hold
address those questions before merging:
1. can the the value of defaultRuntimeClass be amended as a day two operations when we have existing VMIs?
2. if so, what's the impact on existing VMIs?
3. is it going to affect live migration with the target pod getting configured with the new value for defaultRuntimeClass?

File: api/v1beta1/hyperconverged_types.go [hunk 1]
```diff
@@ -112,6 +112,13 @@ type HyperConvergedSpec struct {
 	// +optional
 	DefaultCPUModel *string `json:"defaultCPUModel,omitempty"`
 
+	// DefaultRuntimeClass defines a cluster default for the RuntimeClass to be used for VMIs pods if not set there.
+	// Default RuntimeClass can be changed when kubevirt is running.
+	// TODO: what's the impact on existing VMIs ???
+	// TODO: is it going to affect live migrations ???
```
Comment (tiraboschi) [inline api/v1beta1/hyperconverged_types.go:118]: @acardace @vladikr 
can you please help on this?

**Comment by vladikr** [comment]
> /hold address those questions before merging:
> 
> 1. can the the value of defaultRuntimeClass be amended as a day two operations when we have existing VMIs?

I think that this should be editable.
Runtimeclasses created by the performance addon operator may have a random suffix. If we are going to use this field to set such runtime class we should allow changing it when/if the profile changes. 

> 2. if so, what's the impact on existing VMIs?

The runtime class is being set on pod so as long as the pod is running it will not change.


> 3. is it going to affect live migration with the target pod getting configured with the new value for defaultRuntimeClass?

If this field is changed then the target pod will run with a different runtimeclass. runtime classes configuration can be not suitable to run VMs, however, since it's the cluster admins who set the defaultclass I would trust that it's being done out of certain knowledge.

**Comment by tiraboschi** [comment]
hco-e2e-kv-smoke-gcp  lane succeeded.
/override ci/prow/hco-e2e-kv-smoke-azure

**Comment by tiraboschi** [comment]
hco-e2e-upgrade-index-aws lane succeeded.
/override ci/prow/hco-e2e-upgrade-index-azure

**Comment by tiraboschi** [comment]
hco-e2e-upgrade-index-sno-aws lane succeeded.
/override ci/prow/hco-e2e-upgrade-index-sno-azure

**Pull request merged**